src/index.ts

```diff
@@ -1,19 +1,16 @@
 import {DependencyList, useEffect, useState} from 'react'
 
-export function useAsyncMemo<T>(factory: () => Promise<T> | undefined | null, deps: DependencyList, initial: T = undefined): T {
-  const [val, setVal] = useState<T>(initial)
+export function useAsyncMemo<T>(factory: (...args: any) => Promise<T>, deps?: DependencyList, initial?: T) {
```

Comment:
Why changing the type of `factory`? It is not expected to have a `factory` function with arguments. Just like the type of `useMemo`:

```ts
function useMemo<T>(factory: () => T, deps: DependencyList | undefined): T;
```

Reply:
For example try to set `fetch` as factory

Reply:
You will get

Reply:
The type checking error is reasonable. `fetch` has `input` argument which is required. You shouldn't pass it to `useAsyncMemo` cause `useAsyncMemo` will call `factory` with no arguments.
Take this for example: If you pass `(foo?: number) => {...}` as `factory`, it's ok. But `(foo: number) => {...}` is not.

Reply:
Sorry for the late reply...

Reply:
Part of my code was in pure js, this is the reason